pkg/autodetect/main.go

```diff
@@ -243,7 +243,10 @@ func (b *Background) cleanDeployments(ctx context.Context) {
 
 	// check deployments to see which one needs to be cleaned.
 	for _, dep := range deployments.Items {
-		if instanceName, ok := dep.Annotations[inject.Annotation]; ok {
+		if dep.Labels == nil {
```

Comment:
why the check? Was it causing NPE somewhere?

Reply:
At some point it throws me a NPE on the tests, but it seems like this is not necessary in the latest changes.